src/object/ispy.cpp

```diff
@@ -86,16 +87,16 @@ Ispy::collision(GameObject& , const CollisionHit& )
 }
 
 void
-Ispy::update(float )
+Ispy::update(float dt_sec)
 {
 
   if (state == ISPYSTATE_IDLE) {
     // check if a player has been spotted
     Vector eye = m_col.m_bbox.get_middle();
     if (dir == Direction::LEFT) eye = Vector(m_col.m_bbox.get_left(), m_col.m_bbox.get_middle().y);
-    if (dir == Direction::RIGHT) eye = Vector(m_col.m_bbox.get_right(), m_col.m_bbox.get_middle().y);
-    if (dir == Direction::UP) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_top());
-    if (dir == Direction::DOWN) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_bottom());
+    else if (dir == Direction::RIGHT) eye = Vector(m_col.m_bbox.get_right(), m_col.m_bbox.get_middle().y);
+    else if (dir == Direction::UP || (dir == Direction::DOWN && m_flip != NO_FLIP)) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_top());
+    else if (dir == Direction::DOWN) eye = Vector(m_col.m_bbox.get_middle().x, m_col.m_bbox.get_bottom());
```

Comment:
Switch-case? :wink:

Reply:
Okay, the ispy code is a mess. In the end, I decided to rewrite most of it. The `facing-down` property is useless, direction can be determined from, well, the _direction_. I'll make the direction be determined automatically, and I will add sprite actions for facing up. Instead of making the ispy flip on levelflip, I'll change its direction from DOWN to UP (and vice versa) and reload the sprite action. Please check the code quality again when I push it. I'll also probably delete the old images since they aren't used by the sprite or anywhere in the game, they just weren't removed then the new ispy images were added.